pkg/daemon/update.go

```diff
@@ -423,10 +427,32 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 	}
 
 	if dn.nodeWriter != nil {
-		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStaged", "Changes to OS staged")
+		// We send out the event OSUpdateStaged synchronously to ensure it is recorded.
+		// Remove this when we can ensure all events are sent before exiting.
+		ref, err := reference.GetReference(scheme.Scheme, dn.node)
+		if err == nil {
+			t := metav1.Time{Time: time.Now()}
+			event := &corev1.Event{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%v.%x", ref.Name, t.UnixNano()),
```

Comment:
Since this is the only place we use `ref`, might as well just fetch node name directly off the object and not have to go through the additional `getReference`